### PR TITLE
Updated CNS installation instructions

### PR DESCRIPTION
**Pull request by amjjbonvin**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and that you comply with the following criteria:

- [ ] You have sticked to Python. Please talk to us before adding other programming languages to HADDOCK3
- [x] Your PR is about CNS
- [ ] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [ ] You structured the code into small functions as much as possible. You can use classes if there is a (state) purpose
- [ ] Your code follows our coding style
- [ ] You wrote tests for the new code
- [ ] `tox` tests pass. *Run `tox` command inside the repository folder*
- [ ] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [ ] PR does not add any dependencies, unless permission granted by the HADDOCK team
- [ ] PR does not break licensing
- [x] Your PR is about writing documentation for already existing code :fire:
- [ ] Your PR is about writing tests for already existing code :godmode:

---

"This PR improves the readablity of the CNS installation instructions and also changes the folder structure of the patched CNS files" @rvhonorato

### CODE REVIEW

**Comment by rvhonorato** [comment]
ping @VGPReys @AnnaKravchenko

File: docs/CNS.md [hunk 1]
```diff
@@ -3,100 +3,123 @@
...
+sudo apt-get install gcc gfortran csh
+```
+
+## 1. Downloading CNS
+
+Downloading CNS requires a license, which academic users can [request for free](http://cns-online.org).
+
+To Download CNS, go to their [website](http://cns-online.org) and  click on the `Download` menu and fill the form. You will be sent a password for the download via e-mail - keep in mind only education e-mails are allowed for this.
+
+You will receive a login/password in the e-mail, once you have those go to the [download page](http://cns-online.org/download/v1.3/cns_solve_1.3_all.tar.gz) and use the provided login/password to download the CNS package.
```
Comment (VGPReys) [inline docs/CNS.md:18]: While here the link is directly for the downloading of the 1.3 version, on the email we get, the link is pointing to [http://cns-online.org/download/](http://cns-online.org/download/).
This page contains multiple download links.
But it is not mentioned that we should download the v1.3.
Screenshot 2024-05-27 at 10-41-57 Crystallography   NMR System

File: docs/CNS.md [hunk 2]
```diff
@@ -3,100 +3,123 @@
...
 
-HADDOCK requires some adjustments in CNS to function properly.
-The set of routines to add is provided in the haddock3 repository under [`varia/cns1.3/`](/varia/cns1.3/README.md).
-Copy all of these files into the CNS source directory (`/PATH/TO/cns_solve_1.3/source/`), for example:
+Check if the substitution worked:
 
 ```bash
-cp ~/software/haddock3/varia/cns1.3/* source/
+sed -n 18p cns_solve_env
 ```
```
Comment (VGPReys) [inline docs/CNS.md:59]: I would add the expected STDOUT here:
```	    setenv CNS_SOLVE '/Absolute/path/where/cns_solve/is/cns_solve_1.3'``` ?

File: docs/CNS.md [hunk 3]
```diff
@@ -3,100 +3,123 @@
...
 
-Make sure you are inside the main `cns_solve_1.3` directory.
+> ❗ You cannot use CNS in HADDOCK without these patched files❗
 
-If a suitable compiler is installed on your system, the following command should start the compilation of CNS:
+Copy the CNS files distributed with HADDOCK3 to the CNS directory:
 
 ```bash
-make install
+\cp -r ~/software/haddock3/varia/cns1.3/[bis]*  ~/software/cns_solve_1.3/
```
Comment (VGPReys) [inline docs/CNS.md:69]: why is there a `\` sign before the `cp` command ?

Comment (VGPReys) [reply docs/CNS.md:69]: Also the `[bis]` statement is quite confusing

**Comment by amjjbonvin** [comment]
> why is there a \ sign before the cp command ?
> 

Basic Linux, with the \ is uses the basic command, without any option possibly defined in an alias.

**Comment by amjjbonvin** [comment]
> Also the [bis] statement is quite confusing
> 
Again, Linux shell selection syntax: It will select any file or directory starting with a: b, i or s

File: docs/CNS.md [hunk 4]
```diff
@@ -3,100 +3,123 @@
...
 
-If you are using sh/bash, edit `.cns_solve_env_sh`:
+You can do that either by opening this file in a code editor such as VScode or vim, or by running the command below:
 
-```
-# CHANGE THE NEXT LINE TO POINT TO THE LOCATION OF THE CNSsolve DIRECTORY
-
-        CNS_SOLVE=/PATH/TO/cns_solve_1.3
+```bash
+sed -i 's@_CNSsolve_location_@'"$PWD"'@' cns_solve_env
```
Comment (AnnaKravchenko) [inline docs/CNS.md:49]: Should we add ` sed -i '' 's@_CNSsolve_location_@'"$PWD"'@‘ cns_solve_env ` as an alternative to installing `gnu-sed` on Mac? 
But also idk if it works in general or only ‘on my machine’

File: docs/CNS.md [hunk 5]
```diff
@@ -3,100 +3,123 @@
...
 Once you have [cloned the HADDOCK3 repository](INSTALL.md), please follow these instructions to compile a CNS executable that can be used to run HADDOCK.
 
-## 1 Download CNS
+Before starting, make sure you download the system requirements, they are `gcc`, `gfortran` and `csh`. Luckily they are easily available from the package manager, so if you are using Ubuntu (for example):
 
-Downloading CNS requires a license, which academic users can [request for free](http://cns-online.org) (click on the `Download` menu and fill the form). You will be sent a password for the download via e-mail.
-Using the link provided in the same e-mail, download CNS ("Source installer (compilation required on your system)") to the directory in which you want to install it (for example `software`). Navigate to this directory, then uncompress and extract the archive:
+```bash
+sudo apt-get install gcc gfortran csh
+```
```
Comment (VGPReys) [inline docs/CNS.md:10]: My trouble-shoot was that `gfortran` was not in my PATH.
I had to edit my `.bash_profile`, and add `export PATH="/path/homebrew/bin:$PATH"`.
And make sure the command `which gfortran` was successful

**Review by VGPReys — APPROVED** [review]
Amazing job Alex !
Managed to compile CNS on my MAC M2 !

File: docs/CNS.md [hunk 6]
```diff
@@ -3,100 +3,123 @@
...
 
-Lastly, CNS needs to be loaded in the current shell before it can be used.
+> You might get something like `unknown-...`, in that case installation is still possible, however it means the compiler options might not be properly defined - check the Troubleshooting section
 
-For csh/tcsh:
+Several `Makefile` headers are already provided - here we will focus on the ones that use `gfortran` as the compiler.
 
 ```bash
-source cns_solve_env
+ls ~/software/cns_solve_1.3/instlib/machine/supported/*/Makefile*gfortran*
```
Comment (AnnaKravchenko) [inline docs/CNS.md:96]: Is this really necessary? 
Don’t see a use of listing makefiles, unless user is instructed to check that a specific file is present etc

**Review by AnnaKravchenko — APPROVED** [review]
Very cool, managed to compile CNS all by myself for the first time!
There are several minor suggestions + typos, otherwise it’s perfect.

File: docs/CNS.md (hunk 5 above)
Comment (AnnaKravchenko) [reply docs/CNS.md:10]: yeah, I suggest adding smth like “make sure gfortran is installed properly via `which gfortran`"

File: docs/CNS.md [hunk 7]
```diff
@@ -82,21 +90,17 @@ ls ~/software/cns_solve_1.3/instlib/machine/supported
...
 ```
 
 > You might get something like `unknown-...`, in that case installation is still possible, however it means the compiler options might not be properly defined - check the Troubleshooting section
 
-Several `Makefile` headers are already provided - here we will focus on the ones that use `gfortran` as the compiler.
-
-```bash
-ls ~/software/cns_solve_1.3/instlib/machine/supported/*/Makefile*gfortran*
-```
+Several `Makefile` headers are already provided - here we will use on the one that uses `gfortran` as the compiler.
```
Comment (amjjbonvin) [inline docs/CNS.md:101]: typo: on the one -> only the one

**Comment by amjjbonvin** [comment]
Does work on my Mac under tcsh, but not under bash…

And there are some character issues for ‘ and #

> Should we add sed -i '' 's@_CNSsolve_location_@'"$PWD"'@‘ cns_solve_env as an alternative to installing gnu-sed on Mac?
> But also idk if it works in general or only ‘on my machine’

**Comment by AnnaKravchenko** [comment]
Yeah, then let’s just keep it as is. After all it’s trivial to make changes manually. 

> Does work on my Mac under tcsh, but not under bash… And there are some character issues for ‘ and #
> […](#)
> Should we add sed -i '' 's@_CNSsolve_location_@'"$PWD"'@‘ cns_solve_env as an alternative to installing gnu-sed on Mac? But also idk if it works in general or only ‘on my machine’

**Pull request merged**